### PR TITLE
Feature/csr 24 - Nationality, Immigration and Employment Restrictions

### PR DESCRIPTION
**Pull request by fantasticjamieburns**

### CODE REVIEW

**Comment by fantasticjamieburns** [comment]
This PR is dependant on https://github.com/TransformCore/CSR-Fast-Stream-backend/pull/8 being deplyed first.

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Applicant
+     */
+    public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: setId isn't required on Doctrine entities

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Address.php [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Address
+     */
+    public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Address.php]: setId isn't required on Doctrine entities

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Eligibility
+     */
+    public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php]: setId isn't required on Doctrine entities

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Nationality.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return Nationality
+     */
+    public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Nationality.php]: setId isn't required on Doctrine entities

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/PhoneNumber.php [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     *
+     * @return PhoneNumber
+     */
+    public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/PhoneNumber.php]: setId isn't required on Doctrine entities

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Registration.php [hunk 6]
```diff
@@ -83,7 +57,7 @@ public function getId()
     /**
      * @param int $id
      *
-     * @return User
+     * @return Registration
      */
     public function setId($id)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Registration.php]: setId isn't required on Doctrine entities

**Comment by eddiejaoude** [comment]
Created this ticket for Type in domain model https://github.com/TransformCore/CSR-Fast-Stream-Domain-Model/issues/17

File: src/TransformCore/Bundle/CsrFastStreamBundle/Persistence/ApplicantsRepository.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    private $managerRegistry;
+
+    /**
+     * @var
+     */
+    private $userManager;
+
+    /**
+     * @param \Doctrine\Common\Persistence\ManagerRegistry $managerRegistry
+     */
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Persistence/ApplicantsRepository.php]: @param missing for $userManager

File: src/TransformCore/Bundle/CsrFastStreamBundle/Persistence/ApplicantsRepository.php [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+{
+    const ENTITY_CLASS = 'TransformCore\Bundle\CsrFastStreamBundle\Entity\Applicant';
+
+    /**
+     * @var \Doctrine\Common\Persistence\ManagerRegistry
+     */
+    private $managerRegistry;
+
+    /**
+     * @var
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Persistence/ApplicantsRepository.php]: @var incomplete

File: src/TransformCore/Bundle/CsrFastStreamBundle/DataFixtures/ORM/LoadApplicantData.php [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        $this->addStandardApplicant('Five');
+        $this->addStandardApplicant('Six');
+        $this->addStandardApplicant('Seven');
+        $this->addStandardApplicant('Eight');
+        $this->addStandardApplicant('Nine');
+
+        $manager->flush();
+    }
+
+    private function addStandardApplicant(
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/DataFixtures/ORM/LoadApplicantData.php]: Add missing docblock (Dave to complete)

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+ *
+ * @package TransformCore\CsrFastStream\DomainModel
+ */
+class Eligibility
+{
+
+    /**
+     * @var int|null
+     */
+    protected $id = null;
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php]: No need to assign null to $id

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+    /**
+     * @var Nationality|null
+     */
+    protected $presentNationality = null;
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Eligibility.php]: No need to assign null to $presentNationality

File: src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * Class Applicant
+ * @package TransformCore\Bundle\CsrFastStreamBundle\Form\Validation
+ */
+class Applicant
+{
+    public static function validate(FormInterface $form)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php]: Add docblock (Dave to complete)

File: src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        $groups = array('CsrProfile');
+
+        $groups[] = self::password($form);
+
+        return array_filter(
+            array_unique($groups)
+        );
+    }
+
+    public static function password(FormInterface $form)
```
Comment (knasher) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php]: Add docblock (Dave to complete)

File: test/Spec/TransformCore/Bundle/CsrFastStreamBundle/Entity/AddressSpec.php [hunk 14]
```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Spec\TransformCore\Bundle\CsrFastStreamBundle\Entity;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class AddressSpec extends ObjectBehavior
+{
+    function it_is_initializable()
```
Comment (knasher) [inline test/Spec/TransformCore/Bundle/CsrFastStreamBundle/Entity/AddressSpec.php]: Add docblock (Dave to complete)

File: test/Spec/TransformCore/Bundle/CsrFastStreamBundle/Entity/AddressSpec.php [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        $this->shouldHaveType('TransformCore\Bundle\CsrFastStreamBundle\Entity\Address');
+    }
+
+    function it_should_set_then_get_id()
```
Comment (knasher) [inline test/Spec/TransformCore/Bundle/CsrFastStreamBundle/Entity/AddressSpec.php]: Add docblock (Dave to complete)

File: src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php (hunk 13 above)
Comment (fantasticjamieburns) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php]: Any reason for this to be public?

Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Form/Validation/Applicant.php]: No, fair point

**Comment by knasher** [comment]
Reviewed: Comments left, small changes made, no blockers

File: src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php (hunk 1 above)
Comment (eddiejaoude) [inline src/TransformCore/Bundle/CsrFastStreamBundle/Entity/Applicant.php]: Once this is all merged back in (including F/E & B/E repos), we can look at removing the setters for the IDs & run all test suites again

**Pull request merged**